### DISCUSSION TITLE
Jetbrains: opt-in per project

### DISCUSSION THREAD

**Discussion by leoluk** [community/community#6955] [Copilot Conversations]
It should be possible to configure the Jetbrains plugins such that it needs to be explicitly enabled for each project.

To my understanding, when using Copilot, parts of the source code being edited is being sent to GitHub and retained, possibly indefinitely:

> For example, when you edit files with the GitHub Copilot extension/plugin enabled, **file content snippets**, suggestions, and any modifications to suggestions will be shared with **GitHub, Microsoft, and OpenAI**, and used for diagnostic purposes to improve suggestions and related products.

(https://docs.github.com/en/github/copilot/github-copilot-telemetry-terms)

Even though the data isn't used to train the model, it still makes Copilot unsuitable for use with any proprietary code, like work or customer projects, which are typically subject to strict policies that disallow sharing with third parties. Therefore, it should be possible to enable it on a case-by-case basis (like open source projects) while defaulting to "off".

**Comment by jansorg** [comment]
Hey @leoluk :smile: 
Currently, there's only a setting to turn off automatic completions, https://github.com/github/copilot-docs/blob/main/docs/jetbrains/gettingstarted.md#enabling
That's currently an application-wide setting and not for each project, so it's probably not good enough yet for your use case, though.

**Comment by alexei-osipov** [comment]
I have a situation when I can't use Copilot on some projects (due to legal reasons) and yet have some other (open source) projects where Copilot is OK. So without possibility to enable Copilot on per-project basis it becomes very troublesome to use Copilot at all.

**Comment by linux-china** [comment]
Could Copilot plugin detect `.vscode/settings.json` file in proect to check Copilot enable or not, and of course Copilot is enabled by default if without ` "copilot.enable": false`.   I think settings.json is good for JetBrains IDEs and VS Code.  

```json
{
   "copilot.enable": false
   "deno.enable": true
}
```
